src/open_inwoner/accounts/tests/test_oidc_views.py

```diff
@@ -159,13 +285,63 @@ def test_new_user_is_created_when_new_email(
         )
 
         self.assertRedirects(
-            callback_response, reverse("pages-root"), fetch_redirect_response=False
+            callback_response, reverse("admin:index"), fetch_redirect_response=True
         )
-        new_user = User.objects.filter(email="new_user@example.com")
 
-        self.assertTrue(new_user.exists())
-        self.assertEqual(new_user.get().oidc_id, "some_username")
-        self.assertEqual(new_user.get().login_type, LoginTypeChoices.oidc)
+        new_user = User.objects.get(email="new_user@example.com")
+
+        self.assertEqual(new_user.oidc_id, "some_username")
+        self.assertEqual(new_user.login_type, LoginTypeChoices.oidc)
+        self.assertEqual(new_user.is_staff, True)
+
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.get_userinfo")
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.store_tokens")
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.verify_token")
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.get_token")
+    @patch(
+        "mozilla_django_oidc_db.mixins.OpenIDConnectConfig.get_solo",
+        return_value=OpenIDConnectConfig(id=1, enabled=True, make_users_staff=False),
+    )
+    @patch(
+        "open_inwoner.configurations.models.SiteConfiguration.get_solo",
+        return_value=SiteConfiguration(openid_display=OpenIDDisplayChoices.regular),
+    )
+    def test_new_regular_user_is_created_when_new_email(
+        self,
+        mock_config_get_solo,
+        mock_get_solo,
+        mock_get_token,
+        mock_verify_token,
+        mock_store_tokens,
+        mock_get_userinfo,
+    ):
+        # set up a user with a non existing email address
+        mock_get_userinfo.return_value = {
+            "email": "new_user@example.com",
+            "sub": "some_username",
+        }
+        UserFactory.create(email="existing_user@example.com")
+        session = self.client.session
+        session["oidc_states"] = {"mock": {"nonce": "nonce"}}
+        session.save()
+        callback_url = reverse("oidc_authentication_callback")
+
+        self.assertFalse(User.objects.filter(email="new_user@example.com").exists())
```

Comment:
I personally feel this line is not strictly necessary, I'd assume we could trust the database was cleared. Although the line clearly shows our expectations we don't really do that in general.

Reply:
You're right, it does indeed indicate that previously we didn't have isolated tests, so I'll remove it as we're moving away from using `setUpTestData`